.github/workflows/nightly.yml

```diff
@@ -0,0 +1,37 @@
+---
+# A schedule only runs on branches that match include rules from _both_ `main`
+# and the branch itself.  On `main`, we blanket include all branches that might
+# want to be enabled, then particular branches can override it to exclude
+# themselves by removing themselves from the trigger list.  For example, old
+# stable branches can remove `stable/*` from their copy of this file once they
+# reach their end-of-life.
+name: Nightly Main tests
+on:
+  # remove when proven to work
+  # pull_request:
+  schedule:
+    - cron: "20 6 * * *"
```

Comment:
How did you settle on this time?

Reply:
This was the same time used in Azure. Do you know if there's a way of making it only run if the code has changed? The same way `schedule/always` worked in Azure.

Reply:
I don't see anything in the docs for that in the trigger conditions. You could probably hack it together with runtime checks and storing the last sha1 run somewhere. But I wouldn't worry about that. I was more just curious if you picked this time or just copied it. It's reasonable to run things at 6:20 UTC, activity is typically pretty slow then.